Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,38 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  "$HOMEBREW_GIT" "$@"
+}
+
 which_git() {
-  local which_git
   local active_developer_dir
 
-  which_git="$(which git 2>/dev/null)"
-  if [[ -n "$which_git" && "/usr/bin/git" = "$which_git" ]]
+  if [[ -n "$HOMEBREW_GIT" && -x "$HOMEBREW_GIT" ]]
+  then
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  if [[ -n "$GIT" && -x "$GIT" ]]
+  then
+    HOMEBREW_GIT="$GIT"
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  HOMEBREW_GIT="$(which git 2>/dev/null)"
```

Comment:
> How about creating a Library/Utils/git.sh file which is a bash equivalent of Library/Homebrew/utils/git.rb and only source it when needed by bash commands.

If that's the case: I'd rather leave it as-is for now rather than splitting it up before it's necessary to do so. Let's wait until we have more Bash commands :grinning: